ui/pages/confirmations/components/confirm/info/shared/redesigned-gas-fees/fees-details.tsx

```diff
@@ -0,0 +1,110 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import React, { Dispatch, SetStateAction } from 'react';
+import { useSelector } from 'react-redux';
+import { useI18nContext } from '../../../../../../../hooks/useI18nContext';
+import { currentConfirmationSelector } from '../../../../../../../selectors';
+import { useSupportsEIP1559 } from '../../hooks/useSupportsEIP1559';
+import { EditGasFeesRow } from './edit-gas-fees-row';
+import { FeesRow } from './fees-row';
+import { GasTimings } from './gas-timings';
+import { LayerFeesRow } from './layer-fees-row';
+import { useFeeCalculations } from './use-fees-calculations';
+
+export const FeesDetails = ({
+  gasEstimate,
+  maxFeePerGas,
+  maxPriorityFeePerGas,
+  setShowCustomizeGasPopover,
+  showAdvancedDetails,
+}: {
+  gasEstimate: string;
+  maxFeePerGas: string;
+  maxPriorityFeePerGas: string;
+  setShowCustomizeGasPopover: Dispatch<SetStateAction<boolean>>;
+  showAdvancedDetails: boolean;
+}) => {
+  const t = useI18nContext();
+
+  const transactionMeta = useSelector(
+    currentConfirmationSelector,
+  ) as TransactionMeta;
+
+  const layer1GasFee = transactionMeta?.layer1GasFee ?? null;
+  const hasLayer1GasFee = layer1GasFee !== null;
+
+  const { supportsEIP1559 } = useSupportsEIP1559(transactionMeta);
+
+  const {
+    currentCurrencyEstimatedFeeFromBreakdown,
+    currentCurrencyEstimatedFee,
+    nativeCurrencyEstimatedFeeFromBreakdown,
+    nativeCurrencyEstimatedFee,
+    currentCurrencyL1Fee,
+    nativeCurrencyL1Fee,
+    currentCurrencyL2Fee,
+    nativeCurrencyL2Fee,
+    currentCurrencyMaxFee,
+    nativeCurrencyMaxFee,
+  } = useFeeCalculations(
```

Comment:
I'm not against minor property passing, but if we're repeatedly passing many properties like this, could we encapsulate some of these values into their own hooks instead?